src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  sides: number[];
+
+  constructor(color: Color, a: number, b: number, c: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again

Reply:
I made corrections but I have no idea if that was the point... typescript is black magic to me ;///

Reply:
So imagine a scenario: we have 1 interface for example: `Figure`, and it's used in 10 different classes in our project, let's say you want to change color type and add 'purple', in this pattern you could do that without worrying about forgetting some classes, TS will remind you, you're forgetting something 😄 

Your solution is also valid, but it's better practice to infer as many types as possible, and since you're implementing `Figure` we're infering from `Figure`

@MateuszSlezak